pkg/query-service/app/queryBuilder/functions.go

```diff
@@ -281,6 +281,21 @@ func ApplyFunction(fn v3.Function, result *v3.Result) *v3.Result {
 		return funcMedian5(result)
 	case v3.FunctionNameMedian7:
 		return funcMedian7(result)
+	case v3.FunctionNameTimeShift:
+		shift, ok := fn.Args[0].(float64)
+		if !ok {
+			return result
+		}
+		return funcTimeShift(result, shift)
+	}
+	return result
+}
+
+func funcTimeShift(result *v3.Result, shift float64) *v3.Result {
```

Comment:
this function cannot be used to compare to a previous interval right? Eg, I want to compare today's error rate with yesterday's error rate.

Reply:
No, it can be used and that is the goal of this function. Here https://github.com/SigNoz/signoz/pull/4445/files#diff-998df71bdc12a1398abbbdabd8dbab8fce344d7d59e1653dd2f14b9ac4e8911fR24-R27 we shift the time by x amount to get the data for past using the query. Since the result time will be in past data but the UI chart will have the current window, we adjust the timeshifted query to the current time window.

Reply:
right. Got it